File: 1616-minimum-difference-between-largest-and-smallest-value-in-three-moves/1616-minimum-difference-between-largest-and-smallest-value-in-three-moves.java
class Solution {
    public int minDifference(int[] nums) {
        Arrays.sort(nums);
        if(nums.length <= 4) return 0;
        int res = Integer.MAX_VALUE;
        for(int l = 0 ; l < 4; l++){
            int r = nums.length - 4 + l;
            res = Math.min(res, nums[r] - nums[l]);
        }
        return res;
    }
}

File: 1616-minimum-difference-between-largest-and-smallest-value-in-three-moves/1616-minimum-difference-between-largest-and-smallest-value-in-three-moves.py
class Solution:
    def minDifference(self, nums: List[int]) -> int:
        if len(nums) <= 4:
            return 0
        min_four = sorted(heapq.nsmallest(4, nums))
        max_four = sorted(heapq.nlargest(4, nums))  
        res= float("inf")
        for i in range(4):
            res = min(res, max_four[i] - min_four[i])
        return res

        # nums.sort()
        # res = float("inf")
        # for l in range(4):
        #     r = len(nums) - 4 + l
        #     res = min(res, nums[r] - nums[l])
        # return res
